keymanager-lib/src/kdf.rs

```diff
@@ -683,4 +693,133 @@ mod tests {
+            },
+            // Different master secret.
+            TestVector {
+                master_secret: "54083e90f05860653161bc1575765b3311f6a55d1cab84919cb1e2b02c8351ec",
+                seed: "2a9d51ed0380d11b26255ecc894dfc3aa48196c9b0da463a3073ea820a65a090",
+                kdf_custom: &RUNTIME_KDF_CUSTOM,
+                xof_custom: &RUNTIME_XOF_CUSTOM,
+                sk: "b8092ab767ddc47cb56807d4d1fd0e66eae0b02e289d1e38d4dfa900619edc47",
+                pk: "6782329a12e821697f1d7cc44ba5644a2d81bd827fe09208549296be51da8b66",
+            },
+            // Different seed.
+            TestVector {
+                master_secret: "3fa39161fe106fb770503558dad41bea3221888c09477d864507e615094f5d38",
+                seed: "6cc9f7dfe776c7e531331eaf5fd8717d638cae65a4ebb0d6f128b1873f0f9c22",
+                kdf_custom: &RUNTIME_KDF_CUSTOM,
+                xof_custom: &RUNTIME_XOF_CUSTOM,
+                sk: "f0d5bf36b424f7b168fa37e1a1407f191a260dda149d78676d4ca077b1e8614c",
+                pk: "9099b1e19c482927eb2785c77b34ef2c4b95f812e089e3b2cd6f2a0a743b7561",
+            },
+            // Different kdf custom.
+            TestVector {
+                master_secret: "3fa39161fe106fb770503558dad41bea3221888c09477d864507e615094f5d38",
+                seed: "2a9d51ed0380d11b26255ecc894dfc3aa48196c9b0da463a3073ea820a65a090",
+                kdf_custom: &EPHEMERAL_KDF_CUSTOM,
+                xof_custom: &RUNTIME_XOF_CUSTOM,
+                sk: "18789f16d8a8fbd75f529f0a1de3e95469eb537c283dc66eb296a6681a46c066",
+                pk: "1aa3e6d86c0779afde8a367dbbb025538fb77e410624ece8b25a6be9e1a5170d",
+            },
+            // Different xof custom.
+            TestVector {
+                master_secret: "3fa39161fe106fb770503558dad41bea3221888c09477d864507e615094f5d38",
+                seed: "2a9d51ed0380d11b26255ecc894dfc3aa48196c9b0da463a3073ea820a65a090",
+                kdf_custom: &RUNTIME_KDF_CUSTOM,
+                xof_custom: &EPHEMERAL_XOF_CUSTOM,
+                sk: "e0e6faa3172634c8fdc47554f0541c7141f7ddfe07333b0dab1801bb5dcd755e",
+                pk: "455f877b6c977c51580e947a5a2ae57b4b1ec0a379cf6509be07f10aa88df735",
+            },
+            // Another random vector.
+            TestVector {
+                master_secret: "b573fa13ec26d21a7a2b5117eb124dcab5883eb63ec0a5a9cc0b1d1948bcfc71",
+                seed: "774cbb9c1e8c634e7e136357d9994f65d72fc0e7d72c1e9ad766db1a74e0bb8c",
+                kdf_custom,
+                xof_custom,
+                sk: "70c077d8bd0c5c2be66d08fbacac4f1a996001883baaf62bd933e2f7de390e5f",
+                pk: "2d62567e9061a7cc64df3793fbee65c863862b8f0acc992e2a39e32757f2c743",
+            },
+        ];
+
+        // Values that don't effect key derivation.
+        let checksum = "100246b37912e916e71ff79982b2f62be892ddf1025cde84804f67e7f5713b75";
+        let runtime_id = "8000000000000000000000000000000000000000000000000000000000000000";
```

Comment:
Runtime ID does affect key derivation so it should be part of the test vectors.

Reply:
Runtime ID in kdf struct does not. However, runtime ID in the `LongTermKeyRequest` does at it effects `seed`. But here I only test if kdf returns the same values for the same inputs (seed, kdf, xof).
Another test could be that `LongTermKeyRequest` produces seeds that depend on the runtime ID. I do not test that but I do test in `requests_have_unique_customs` that customs are different which means that kmac and keys will be different.

Will add missing test that requests produce seeds that are dependent on their values (runtime id, key_pair_id, epoch).

Reply:
Oh I see, this was refactored, yeah makes sense to add explicit tests for those requests as well.